### PR TITLE
ipq40xx: convert AVM FRITZ!Repeater 1200 to DSA

### PR DESCRIPTION
**Pull request by PolynomialDivision**
**Only compile tested**

Person who added device: @blocktrron

### CODE REVIEW

**Comment by jmv2009** [comment]
Ethernet is not working.

**Comment by PussAzuki** [comment]
> Ethernet is not working.

Try using swport1 instead of swport5? According to the dts using edma, ethernet-phy@0 is not deleted and is in use.

**Comment by PolynomialDivision** [comment]
@jmv2009 I did a mistake. Fritz!Repeater 1200 is using rgmi-id. I did all necessary changes. Please try again. Thanks for testing!

**Comment by jmv2009** [comment]
Still no ethernet:

- [    1.144156] NET: Registered protocol family 10
- [    1.147465] Segment Routing with IPv6
- [    1.150652] NET: Registered protocol family 17
- [    1.154613] 8021q: 802.1Q VLAN Support v1.8
- [    1.158673] Registering SWP/SWPB emulation handler
- [    1.198755] qca8k-ipq4019 c000000.switch: configuring for fixed/internal link mode
- [    1.198978] qca8k-ipq4019 c000000.switch lan (uninitialized): failed to connect to PHY: -ENODEV
- [    1.205312] qca8k-ipq4019 c000000.switch lan (uninitialized): error -19 setting up PHY for tree 0, switch 0, port 5
- [    1.213925] qca8k-ipq4019 c000000.switch: Link is Up - 1Gbps/Full - flow control rx/tx
- [    1.224513] DSA: tree 0 setup

`

**Comment by PolynomialDivision** [comment]
> Try using swport1 instead of swport5? According to the dts using edma, ethernet-phy@0 is not deleted and is in use.

Whups, I think this is the solution.

**Comment by PolynomialDivision** [comment]
Pushed a fix (hopefully)

**Comment by jmv2009** [comment]
No ethernet; Slightly different error message.

- [    1.139078] NET: Registered protocol family 17
- [    1.143047] 8021q: 802.1Q VLAN Support v1.8
- [    1.147109] Registering SWP/SWPB emulation handler
- [    1.183972] qca8k-ipq4019 c000000.switch: configuring for fixed/internal link mode
- [    1.184156] qca8k-ipq4019 c000000.switch: interface 'rgmii-id' (9) on port 1 is not supported
- [    1.190474] qca8k-ipq4019 c000000.switch lan (uninitialized): failed to connect to PHY: -ENODEV
- [    1.199121] qca8k-ipq4019 c000000.switch lan (uninitialized): error -19 setting up PHY for tree 0, switch 0, port 1
- [    1.207639] qca8k-ipq4019 c000000.switch: Link is Up - 1Gbps/Full - flow control rx/tx
- [    1.218328] DSA: tree 0 setup

**Comment by PolynomialDivision** [comment]
@robimarko can you have a look?

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts [hunk 1]
```diff
@@ -237,6 +241,34 @@
...
+	status = "okay";
+
+	label = "lan";
+	phy-handle = <&ethphy>;
+	phy-mode = "rgmii-id";
+};
+
 &ethphy0 {
 	status = "disabled";
 };
```
Comment (jmv2009) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts]: Removing lines 272 to 275 gets me a bit further (don't disable ethphy0)

- [    1.144970] NET: Registered protocol family 17
- [    1.148808] 8021q: 802.1Q VLAN Support v1.8
- [    1.152995] Registering SWP/SWPB emulation handler
- [    1.188567] qca8k-ipq4019 c000000.switch: configuring for fixed/internal link mode
- [    1.188763] qca8k-ipq4019 c000000.switch: interface 'rgmii-id' (9) on port 1 is not supported
- [    1.195227] qca8k-ipq4019 c000000.switch: Link is Up - 1Gbps/Full - flow control rx/tx
- [    1.287205] qca8k-ipq4019 c000000.switch: interface 'rgmii-id' (9) on port 1 is not supported
- [    1.287647] qca8k-ipq4019 c000000.switch lan (uninitialized): validation of rgmii-id with support 0000000,00000000,000062ef and advertisement 0000000,00000000,000062ef failed: -22
- [    1.296461] qca8k-ipq4019 c000000.switch lan (uninitialized): failed to connect to PHY: -EINVAL
- [    1.311208] qca8k-ipq4019 c000000.switch lan (uninitialized): error -22 setting up PHY for tree 0, switch 0, port 1
- [    1.320535] DSA: tree 0 setup

**Review by jmv2009 — CHANGES_REQUESTED** [review]
Don't disable ethphy0...

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts (hunk 1 above)
Comment (PussAzuki) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts]: It looks like you need to switch swport1 back to swport5 as well, and that should be fine.

Comment (jmv2009) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts]: I was trying to follow  [qcom-ipq4029-aruba-glenmorangie.dtsi] in (https://github.com/openwrt/openwrt/pull/4721/files#diff-619dfd8bfc54ca01560fd270dbf277dae30f8c60c8c34b1b9be4eea226d20268). Both ethphy and swport are renumbered to "5". Odly, there would be no offset, such as between "0" and "1". But it didn't work:

- [    1.141193] Registering SWP/SWPB emulation handler
- [    1.178197] qca8k-ipq4019 c000000.switch: configuring for fixed/internal link mode
- [    1.178410] qca8k-ipq4019 c000000.switch lan (uninitialized): failed to connect to PHY: -ENODEV
- [    1.184746] qca8k-ipq4019 c000000.switch lan (uninitialized): error -19 setting up PHY for tree 0, switch 0, port 5
- [    1.193370] qca8k-ipq4019 c000000.switch: Link is Up - 1Gbps/Full - flow control rx/tx
- [    1.204035] DSA: tree 0 setup

Comment (PussAzuki) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts]: Now in your place ethphy is pointing to 5?

Comment (PussAzuki) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts]: you should see qcom-ipq4029-aruba-glenmorangie.dtsi do delete `ethernet-phy@0` and do add `ethphy: ethernet-phy@5`  in essedma driver version, but this model do not do this, so do not just copy it.

I do think after applying this PR, change swport1 back to swport5 and delete disabled node should be enough.

Comment (jmv2009) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts]: Yes, that appears to work. VLAN's now work through lan.2

**Review by jmv2009 — CHANGES_REQUESTED** [review]
Working now. Thanks. It should be at swport5. VLAN not working.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts (hunk 1 above)
Comment (PussAzuki) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts]: https://openwrt.org/docs/guide-user/network/dsa/converting-to-dsa

I know little about VLAN with DSA, but found it, maybe you can try

Comment (jmv2009) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts]: vlans now work through lan.2

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts [hunk 2]
```diff
@@ -237,6 +241,34 @@
...
+	status = "okay";
+};
+
+&switch {
+	status = "okay";
+
+	/delete-property/ psgmii-ethphy;
+};
+
+&swport1 {
```
Comment (jmv2009) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts]: So this should be swport5.

**Comment by PolynomialDivision** [comment]
@jmv2009 I think I applied all the changes you mentioned? Can you check again and give ma a signed-off I can add to commit message?

**Comment by PussAzuki** [comment]
There should be no problem now.

**Comment by PolynomialDivision** [comment]
> There should be no problem now.

Can I also piut your singed off?

**Comment by PussAzuki** [comment]
> Can I also piut your singed off?

That should be fine.

**Comment by PolynomialDivision** [comment]
Hmmm, OpenWrt has real name policy. :S

**Comment by PussAzuki** [comment]
Ah, let's forget about it for now.

**Comment by jmv2009** [comment]
I tried your new version. As expected, it works on a basic level. Thanks. Somehow, when messing around with VLANS the ethernet turned off alltogether and couldn't be  turned back on without reboot. But that's probably shared by many more devices. This patch should be ok.

**Comment by PolynomialDivision** [comment]
@robimarko please merge. ;)

**Comment by jmv2009** [comment]
For reference, here are the kernel log messages:

[    1.188963] qca8k-ipq4019 c000000.switch: configuring for fixed/internal link mode
[    1.189267] qca8k-ipq4019 c000000.switch: Link is Up - 1Gbps/Full - flow control rx/tx
[    1.277179] qca8k-ipq4019 c000000.switch lan (uninitialized): PHY [90000.mdio-1:00] driver [Qualcomm Atheros AR8031/AR8033] (irq=POLL)
[    1.289326] DSA: tree 0 setup

**Comment by robimarko** [comment]
Picked, thanks.

**Pull request closed without merging**